python_modules/libraries/dagster-dg/dagster_dg/context.py

```diff
@@ -105,6 +113,23 @@ def get_code_location_names(self) -> Iterable[str]:
         return [loc.name for loc in sorted((self.root_path / "code_locations").iterdir())]
 
 
+def get_code_location_env_hash(code_location_root_path: Path) -> str:
+    uv_lock_path = code_location_root_path / "uv.lock"
+    if not uv_lock_path.exists():
+        raise DgError(f"uv.lock file not found in {code_location_root_path}")
+    local_components_path = (
+        code_location_root_path
+        / code_location_root_path.name
+        / _CODE_LOCATION_CUSTOM_COMPONENTS_DIR
```

Comment:
I think we should rename this to `_CODE_LOCATION_COMPONENTS_LIB_DIR` as we are going to have custom components outside of the library sometimes